### DISCUSSION TITLE
Shall we remove the loaded_modules in EMQX Enterprise 4.4.X ？

### DISCUSSION THREAD

**Discussion by Rory-Z** [emqx/emqx-operator#294] [Ideas]
Do we need to create the new APIVersion for v1beta4 and remove the loaded_modules field in EMQX Enterprise ?

For now，emqx-operator will create configmap for loaded_modules, and mount it in emqx container, but the problem is, when user update modules in EMQX Dashboard, EMQX will try write something for loaded_modules, and will got erofs error, because the loaded_modules file is read only.

The loaded_modules file is json file in EMQX Enterprise, initially I wanted the user to be able to modify it like a configuration file, but it is more complicated, users hardly ever modify it directly.

So, can we remove the loaded_modules in EMQX Enterprise 4.4.X, because it will cause erofs error in EMQX Running.

![image](https://user-images.githubusercontent.com/16801068/180965735-d9aa158a-79d7-4a51-9814-4c2604efb55b.png)

**Comment by Rory-Z** [comment]
Other idea, in v1beta3 APIVersion, if user didn't set modules, emqx operator will didn't create loaded_modules configMap anymore, if user set modules, emqx operator will create loaded_modules configMap like now

**Comment by ghost** [comment]
why not use a volume to storage all the moduler's configration files, so the emqx can get the configration modify auth.

**Reply by Rory-Z** [reply]
>  initially I wanted the user to be able to modify it like a configuration file

Users can not modify volume file like configuration
